stl_text/datamodule/doc_classification.py

```diff
@@ -27,17 +51,23 @@ def __init__(self, data_path: str = 'glue_sst2_tiny', vocab_path: Optional[str]
         self.text_transform = None
         self.label_transform = None
         self.datasets = {}
+        self.unk_token = '<unk>'
 
     def setup(self, stage):
-        # torchtext's spm_tokenizer and vocab are not pickleable and will fail dataset.map()
-        # tokenizer = spm_tokenizer(download_from_url(PRETRAINED_SP_MODEL["text_unigram_25000"]))
-        # vocab = build_fairseq_vocab("vocab_tiny.txt")
+        tokenizer = sentencepiece_tokenizer(download_from_url(PRETRAINED_SP_MODEL["text_unigram_25000"])).to_ivalue()
+        if self.vocab_path:
+            path_manager = PathManager()
+            with path_manager.open(self.vocab_path, "r",encoding='utf-8') as f:
+                vocab = build_vocab_from_text_file(f,torch.jit.script(tokenizer))
+        else:
+            vocab = Vocab(VocabPybind([self.unk_token],self.unk_token))
+
+        vocab = vocab.to_ivalue()
```

Comment:
I'd add [TODO] note here. We have a PR to remove the `to_ivalue` func.
https://github.com/pytorch/text/pull/1080